sdks/java/extensions/euphoria/euphoria-core/src/main/java/org/apache/beam/sdk/extensions/euphoria/core/translate/coder/ClassAwareKryoCoder.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.extensions.euphoria.core.translate.coder;
+
+import com.esotericsoftware.kryo.Kryo;
+import com.esotericsoftware.kryo.io.Input;
+import com.esotericsoftware.kryo.io.Output;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.List;
+import java.util.Objects;
+import org.apache.beam.sdk.coders.Coder;
+import org.apache.beam.sdk.coders.CoderException;
+import org.apache.beam.sdk.extensions.euphoria.core.client.functional.VoidFunction;
+import org.objenesis.strategy.StdInstantiatorStrategy;
+
+/**
+ * {@link Coder} which encodes/decodes objects of given {@link Class} through {@link Kryo}.
+ */
+public class ClassAwareKryoCoder<T> extends Coder<T> {
+
+  /**
+   * Initial size of byte buffers in {@link Output}, {@link Input}.
+   */
+  private static final int DEFAULT_BUFFER_SIZE = 4096;
+
+  // this factory needs to be serializable, since the Coder itself is
+  private static final VoidFunction<Kryo> KRYO_FACTORY =
+      () -> {
+        final Kryo instance = new Kryo();
```

Comment:
we should allow user to set required registration flag for kryo as it could have huge performace implications if some classes are not registered (it should be probably enabled by default)

Reply:
Again,  [BEAM-4705](https://issues.apache.org/jira/browse/BEAM-4705).